Realm.Shared/Realm.cs

```diff
@@ -642,7 +660,7 @@ public bool Refresh()
             if (schema?.Type != type)
                 throw new ArgumentException($"The class {type.Name} is not in the limited set of classes for this realm");
             
-            return new RealmResults<T>(this, schema, true);
+            return new RealmResults<T>(this, Metadata[typeof(T).Name], true);
```

Comment:
Since we need the Metadata object here, this could perform a single Metadata lookup:
```csharp
RealmObject.Metadata metadata;
if (!Metadata.TryGetValue(type.Name, out metadata) || metadata.Schema.Type != type)
{
    throw new ArgumentException(...);
}

return new RealmResults<T>(this, metadata, true);
```